packages/machine-extractor/src/MachineExtractResult.ts

```diff
@@ -800,99 +800,67 @@ export class MachineExtractResult {
             ? stateProp.key.value
             : null;
 
+          if (oldName === null) {
+            throw new Error('Could not find the old name of the state');
+          }
```

Comment:
this just moved this error to act as an early~ return and it now compared `oldName` against `null`, instead of checking its truthiness